### PR TITLE
[automation] Auto-update linters version, help and documentation

### PR DESCRIPTION
**Pull request by nvuillam**
[automation] Auto-update linters version, help and documentation

### CODE REVIEW

**Comment by nvuillam** [comment]
## [MegaLinter](https://oxsecurity.github.io/megalinter/beta) status: :warning: [**WARNING**](https://github.com/oxsecurity/megalinter/actions/runs/2874681144)

|    Descriptor    |                                                        Linter                                                        |Files|Fixed|                                  Errors                                  |Elapsed time|
|------------------|----------------------------------------------------------------------------------------------------------------------|-----|----:|--------------------------------------------------------------------------|------------|
|✅ BASH           |[bash-exec](https://oxsecurity.github.io/megalinter/beta/descriptors/bash_bash_exec)                                  |    7|     |                                                                         0|0.01s       |
|✅ BASH           |[shellcheck](https://oxsecurity.github.io/megalinter/beta/descriptors/bash_shellcheck)                                |    7|     |                                                                         0|0.25s       |
|✅ BASH           |[shfmt](https://oxsecurity.github.io/megalinter/beta/descriptors/bash_shfmt)                                          |    7|    0|                                                                         0|0.2s        |
|✅ COPYPASTE      |[jscpd](https://oxsecurity.github.io/megalinter/beta/descriptors/copypaste_jscpd)                                     |yes  |     |no                                                                        |1.77s       |
|✅ DOCKERFILE     |[hadolint](https://oxsecurity.github.io/megalinter/beta/descriptors/dockerfile_hadolint)                              |  103|     |                                                                         0|18.08s      |
|✅ JSON           |[eslint-plugin-jsonc](https://oxsecurity.github.io/megalinter/beta/descriptors/json_eslint_plugin_jsonc)              |   20|    0|                                                                         0|1.86s       |
|✅ JSON           |[jsonlint](https://oxsecurity.github.io/megalinter/beta/descriptors/json_jsonlint)                                    |   18|     |                                                                         0|0.2s        |
|✅ JSON           |[v8r](https://oxsecurity.github.io/megalinter/beta/descriptors/json_v8r)                                              |   20|     |                                                                         0|10.77s      |
|:warning: MARKDOWN|[markdownlint](https://oxsecurity.github.io/megalinter/beta/descriptors/markdown_markdownlint)                        |  281|    0|[**5**](https://github.com/oxsecurity/megalinter/actions/runs/2874681144) |4.11s       |
|✅ MARKDOWN       |[markdown-link-check](https://oxsecurity.github.io/megalinter/beta/descriptors/markdown_markdown_link_check)          |  281|     |                                                                         0|4.63s       |
|✅ MARKDOWN       |[markdown-table-formatter](https://oxsecurity.github.io/megalinter/beta/descriptors/markdown_markdown_table_formatter)|  281|    0|                                                                         0|5.6s        |
|✅ OPENAPI        |[spectral](https://oxsecurity.github.io/megalinter/beta/descriptors/openapi_spectral)                                 |    1|     |                                                                         0|0.73s       |
|:warning: PYTHON  |[bandit](https://oxsecurity.github.io/megalinter/beta/descriptors/python_bandit)                                      |  165|     |[**51**](https://github.com/oxsecurity/megalinter/actions/runs/2874681144)|2.07s       |
|✅ PYTHON         |[black](https://oxsecurity.github.io/megalinter/beta/descriptors/python_black)                                        |  165|    0|                                                                         0|2.97s       |
|✅ PYTHON         |[flake8](https://oxsecurity.github.io/megalinter/beta/descriptors/python_flake8)                                      |  165|     |                                                                         0|1.85s       |
|✅ PYTHON         |[isort](https://oxsecurity.github.io/megalinter/beta/descriptors/python_isort)                                        |  165|    0|                                                                         0|0.46s       |
|✅ PYTHON         |[mypy](https://oxsecurity.github.io/megalinter/beta/descriptors/python_mypy)                                          |  165|     |                                                                         0|89.95s      |
|✅ PYTHON         |[pylint](https://oxsecurity.github.io/megalinter/beta/descriptors/python_pylint)                                      |  165|     |                                                                         0|14.95s      |
|✅ REPOSITORY     |[checkov](https://oxsecurity.github.io/megalinter/beta/descriptors/repository_checkov)                                |yes  |     |no                                                                        |12.9s       |
|✅ REPOSITORY     |[git_diff](https://oxsecurity.github.io/megalinter/beta/descriptors/repository_git_diff)                              |yes  |     |no                                                                        |0.18s       |
|✅ REPOSITORY     |[secretlint](https://oxsecurity.github.io/megalinter/beta/descriptors/repository_secretlint)                          |yes  |     |no                                                                        |9.28s       |
|✅ REPOSITORY     |[trivy](https://oxsecurity.github.io/megalinter/beta/descriptors/repository_trivy)                                    |yes  |     |no                                                                        |20.55s      |
|✅ SPELL          |[cspell](https://oxsecurity.github.io/megalinter/beta/descriptors/spell_cspell)                                       |  661|     |                                                                         0|18.14s      |
|✅ SPELL          |[misspell](https://oxsecurity.github.io/megalinter/beta/descriptors/spell_misspell)                                   |  491|    0|                                                                         0|0.56s       |
|✅ XML            |[xmllint](https://oxsecurity.github.io/megalinter/beta/descriptors/xml_xmllint)                                       |    3|     |                                                                         0|0.0s        |
|✅ YAML           |[prettier](https://oxsecurity.github.io/megalinter/beta/descriptors/yaml_prettier)                                    |   79|    0|                                                                         0|2.58s       |
|✅ YAML           |[v8r](https://oxsecurity.github.io/megalinter/beta/descriptors/yaml_v8r)                                              |   22|     |                                                                         0|24.81s      |
|✅ YAML           |[yamllint](https://oxsecurity.github.io/megalinter/beta/descriptors/yaml_yamllint)                                    |   80|     |                                                                         0|1.14s       |

See errors details in [**artifact MegaLinter reports** on CI Job page](https://github.com/oxsecurity/megalinter/actions/runs/2874681144)

**Comment by nvuillam** [comment]
## [MegaLinter](https://oxsecurity.github.io/megalinter/test-nvuillam-patch) status: :warning: [**WARNING**](https://github.com/oxsecurity/megalinter/actions/runs/2874681142)

|     Descriptor     |                                                               Linter                                                                |Files|Fixed|                                  Errors                                  |Elapsed time|
|--------------------|-------------------------------------------------------------------------------------------------------------------------------------|-----|----:|--------------------------------------------------------------------------|------------|
|✅ BASH             |[bash-exec](https://oxsecurity.github.io/megalinter/test-nvuillam-patch/descriptors/bash_bash_exec)                                  |    7|     |                                                                         0|0.02s       |
|✅ BASH             |[shellcheck](https://oxsecurity.github.io/megalinter/test-nvuillam-patch/descriptors/bash_shellcheck)                                |    7|     |                                                                         0|0.31s       |
|✅ BASH             |[shfmt](https://oxsecurity.github.io/megalinter/test-nvuillam-patch/descriptors/bash_shfmt)                                          |    7|    0|                                                                         0|0.04s       |
|✅ COPYPASTE        |[jscpd](https://oxsecurity.github.io/megalinter/test-nvuillam-patch/descriptors/copypaste_jscpd)                                     |yes  |     |no                                                                        |2.32s       |
|✅ DOCKERFILE       |[hadolint](https://oxsecurity.github.io/megalinter/test-nvuillam-patch/descriptors/dockerfile_hadolint)                              |  103|     |                                                                         0|18.94s      |
|✅ JSON             |[eslint-plugin-jsonc](https://oxsecurity.github.io/megalinter/test-nvuillam-patch/descriptors/json_eslint_plugin_jsonc)              |   20|    0|                                                                         0|2.84s       |
|✅ JSON             |[jsonlint](https://oxsecurity.github.io/megalinter/test-nvuillam-patch/descriptors/json_jsonlint)                                    |   18|     |                                                                         0|0.46s       |
|✅ JSON             |[v8r](https://oxsecurity.github.io/megalinter/test-nvuillam-patch/descriptors/json_v8r)                                              |   20|     |                                                                         0|13.04s      |
|:warning: MARKDOWN  |[markdownlint](https://oxsecurity.github.io/megalinter/test-nvuillam-patch/descriptors/markdown_markdownlint)                        |  281|    0|[**5**](https://github.com/oxsecurity/megalinter/actions/runs/2874681142) |5.05s       |
|✅ MARKDOWN         |[markdown-link-check](https://oxsecurity.github.io/megalinter/test-nvuillam-patch/descriptors/markdown_markdown_link_check)          |  281|     |                                                                         0|5.91s       |
|✅ MARKDOWN         |[markdown-table-formatter](https://oxsecurity.github.io/megalinter/test-nvuillam-patch/descriptors/markdown_markdown_table_formatter)|  281|    0|                                                                         0|7.55s       |
|✅ OPENAPI          |[spectral](https://oxsecurity.github.io/megalinter/test-nvuillam-patch/descriptors/openapi_spectral)                                 |    1|     |                                                                         0|0.95s       |
|:warning: PYTHON    |[bandit](https://oxsecurity.github.io/megalinter/test-nvuillam-patch/descriptors/python_bandit)                                      |  165|     |[**51**](https://github.com/oxsecurity/megalinter/actions/runs/2874681142)|2.67s       |
|✅ PYTHON           |[black](https://oxsecurity.github.io/megalinter/test-nvuillam-patch/descriptors/python_black)                                        |  165|    0|                                                                         0|3.65s       |
|✅ PYTHON           |[flake8](https://oxsecurity.github.io/megalinter/test-nvuillam-patch/descriptors/python_flake8)                                      |  165|     |                                                                         0|2.24s       |
|✅ PYTHON           |[isort](https://oxsecurity.github.io/megalinter/test-nvuillam-patch/descriptors/python_isort)                                        |  165|    0|                                                                         0|0.43s       |
|✅ PYTHON           |[mypy](https://oxsecurity.github.io/megalinter/test-nvuillam-patch/descriptors/python_mypy)                                          |  165|     |                                                                         0|116.04s     |
|✅ PYTHON           |[pylint](https://oxsecurity.github.io/megalinter/test-nvuillam-patch/descriptors/python_pylint)                                      |  165|     |                                                                         0|20.64s      |
|✅ REPOSITORY       |[checkov](https://oxsecurity.github.io/megalinter/test-nvuillam-patch/descriptors/repository_checkov)                                |yes  |     |no                                                                        |16.7s       |
|:warning: REPOSITORY|[devskim](https://oxsecurity.github.io/megalinter/test-nvuillam-patch/descriptors/repository_devskim)                                |yes  |     |[**59**](https://github.com/oxsecurity/megalinter/actions/runs/2874681142)|1.34s       |
|✅ REPOSITORY       |[dustilock](https://oxsecurity.github.io/megalinter/test-nvuillam-patch/descriptors/repository_dustilock)                            |yes  |     |no                                                                        |1.97s       |
|✅ REPOSITORY       |[git_diff](https://oxsecurity.github.io/megalinter/test-nvuillam-patch/descriptors/repository_git_diff)                              |yes  |     |no                                                                        |0.03s       |
|✅ REPOSITORY       |[secretlint](https://oxsecurity.github.io/megalinter/test-nvuillam-patch/descriptors/repository_secretlint)                          |yes  |     |no                                                                        |4.57s       |
|✅ REPOSITORY       |[syft](https://oxsecurity.github.io/megalinter/test-nvuillam-patch/descriptors/repository_syft)                                      |yes  |     |no                                                                        |2.16s       |
|✅ REPOSITORY       |[trivy](https://oxsecurity.github.io/megalinter/test-nvuillam-patch/descriptors/repository_trivy)                                    |yes  |     |no                                                                        |15.89s      |
|✅ SPELL            |[cspell](https://oxsecurity.github.io/megalinter/test-nvuillam-patch/descriptors/spell_cspell)                                       |  684|     |                                                                         0|23.62s      |
|✅ SPELL            |[misspell](https://oxsecurity.github.io/megalinter/test-nvuillam-patch/descriptors/spell_misspell)                                   |  514|    0|                                                                         0|0.53s       |
|✅ XML              |[xmllint](https://oxsecurity.github.io/megalinter/test-nvuillam-patch/descriptors/xml_xmllint)                                       |    3|     |                                                                         0|0.0s        |
|✅ YAML             |[prettier](https://oxsecurity.github.io/megalinter/test-nvuillam-patch/descriptors/yaml_prettier)                                    |   79|    0|                                                                         0|2.7s        |
|✅ YAML             |[v8r](https://oxsecurity.github.io/megalinter/test-nvuillam-patch/descriptors/yaml_v8r)                                              |   22|     |                                                                         0|30.78s      |
|✅ YAML             |[yamllint](https://oxsecurity.github.io/megalinter/test-nvuillam-patch/descriptors/yaml_yamllint)                                    |   80|     |                                                                         0|1.31s       |

See errors details in [**artifact MegaLinter reports** on CI Job page](https://github.com/oxsecurity/megalinter/actions/runs/2874681142)

File: docs/mega-linter-runner.md [hunk 1]
```diff
@@ -56,14 +56,10 @@ Sample `.pre-commit-config.yaml`:
...
+    rev: v5.12.0 # Git tag specifying the hook, not mega-linter-runner, version
     hooks:
-      - id: megalinter-incremental # Faster, less thorough
-        stages:
-          - commit
-      - id: megalinter-full # Slower, more thorough
-        stages:
-          - push
+      - id: megalinter # Faster, less thorough, runs pre-commit by default
+      - id: megalinter-all # Slower, more thorough, runs pre-push by default
```
Comment (Kurt-von-Laven) [inline docs/mega-linter-runner.md:62]: @nvuillam, did the automation revert this portion of #1756 by accident?

Comment (nvuillam) [reply docs/mega-linter-runner.md:62]: image

Meaning you need to update the README of mega-linter-runner so it is propagated to docs/mega-linter-runner.md :)

Comment (nvuillam) [reply docs/mega-linter-runner.md:62]: You can do that in your current pre-commit PR ^^

Comment (Kurt-von-Laven) [reply docs/mega-linter-runner.md:62]: Wow, I should know that by now. Thank you for the reminder! I updated my PR accordingly.

**Pull request merged**